pkg/security/jwt/validator.go

```diff
@@ -34,6 +35,26 @@ func (v *Validator) Parse(token string) (jwt.MapClaims, error) {
 	return c, nil
 }
 
+func (v *Validator) ParseWithContext(ctx context.Context, token string) (jwt.MapClaims, error) {
+	if token == "" {
+		return nil, fmt.Errorf("missing token")
+	}
+
+	jwtKeyfunc := func(token *jwt.Token) (interface{}, error) {
+		return v.keys.GetOrFetchKeyWithContext(ctx, token)
+	}
+
+	t, err := jwt.Parse(token, jwtKeyfunc)
+	if t == nil {
+		return nil, fmt.Errorf("could not parse token: %w", err)
+	}
+	c := t.Claims.(jwt.MapClaims)
```

Comment:
This seems a bit weird, shouldn't it be something like?

```
c, ok := t.Claims.(jwt.MapClaims)
if !ok {
  return error
}

Reply:
fixed